### PR TITLE
Fix morph interpolation

### PR DESCRIPTION
**Pull request by A-Walrus**
# Objective
Fixes: #9898
## Solution
Make morph behave like other keyframes, lerping first between start and end, and then between the current state and the result.

## Changelog
Fixed jerky morph targets

### CODE REVIEW

**Review by mockersf — APPROVED** [review]

File: crates/bevy_animation/src/lib.rs [hunk 1]
```diff
@@ -565,8 +565,17 @@ fn run_animation_player(
     }
 }
 
-/// Update `weights` based on weights in `keyframes` at index `key_index`
-/// with a linear interpolation on `key_lerp`.
+/// Update `weights` based on weights in `keyframe` with a linear interpolation
+/// on `key_lerp`.
+fn lerp_morph_weights(weights: &mut [f32], key_lerp: f32, keyframe: impl Iterator<Item = f32>) {
```
Comment (nicopap) [inline crates/bevy_animation/src/lib.rs:570]: Nit: could we swap the two arguments.
```suggestion
fn lerp_morph_weights(weights: &mut [f32], keyframe: impl Iterator<Item = f32>, key_lerp: f32) {
```

File: crates/bevy_animation/src/lib.rs [hunk 2]
```diff
@@ -565,8 +565,17 @@ fn run_animation_player(
...
+    let zipped = weights.iter_mut().zip(keyframe);
+    for (morph_weight, keyframe) in zipped {
+        let minus_lerp = 1.0 - key_lerp;
+        *morph_weight = (*morph_weight * minus_lerp) + (keyframe * key_lerp);
```
Comment (nicopap) [inline crates/bevy_animation/src/lib.rs:574]: @CGMossa recommended this

```suggestion
        *morph_weight += (keyframe - *morph_weight) * key_lerp;
```

**Review by nicopap — APPROVED** [review]
LGTM

File: crates/bevy_animation/src/lib.rs [hunk 3]
```diff
@@ -707,7 +714,14 @@ fn apply_animation(
...
                     Keyframes::Weights(keyframes) => {
                         if let Ok(morphs) = &mut morphs {
-                            lerp_morph_weights(morphs.weights_mut(), weight, keyframes, step_start);
+                            let target_count = morphs.weights().len();
+                            let morph_start = get_keyframe(target_count, keyframes, step_start);
+                            let morph_end = get_keyframe(target_count, keyframes, step_start + 1);
+                            let result = morph_start
+                                .iter()
+                                .zip(morph_end)
+                                .map(|(a, b)| *a * (1.0 - lerp) + *b * lerp);
```
[the remark this answers is not in the document]
Comment (CGMossa) [reply crates/bevy_animation/src/lib.rs:723]:
```suggestion
                                .map(|(a, b)| *a * + lerp * (*b - *a));
```

**Review by CGMossa — APPROVED** [review]
LGTM

I investigated this with @nicopap; As they attempted to mentor me through solving this.
I learned a lot, and thus able to review this a little bit.

Very impressive work! I still don't fully get it, but it is as far as I got myself!

**Pull request merged**